### PR TITLE
Add EtSum Helper Class and Demo Code

### PR DESCRIPTION
**Pull request by mulhearn**
This is for downstream consumers of the L1T output.  A Helper class for interpreting somewhat confusing new format for the L1T EtSums is provided, along with demonstration code for all of L1T upgrade objects.

### CODE REVIEW

**Comment by mulhearn** [comment]
@davidlange6 this clearly isn't a bugfix, but it will make integrating the downstream code easier if this helper class and demo are available in the release.  Nothing currently in the release depends on any of this code yet.

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
+1

File: DataFormats/L1Trigger/interface/EtSumHelper.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    EtSumHelper(edm::Handle<l1t::EtSumBxCollection> & sum ):sum_(sum) {} // class assumes sum has been checked to be valid. 
+    double MissingEt();
+    double MissingEtPhi();
+    double MissingHt();
+    double MissingHtPhi();
+    double TotalEt();
+    double TotalHt();
+
+  private:
+    edm::Handle<l1t::EtSumBxCollection> & sum_;
```
Comment (davidlange6) [inline DataFormats/L1Trigger/interface/EtSumHelper.h]: @mulhearn - could this be const?

File: L1Trigger/L1TGlobal/src/Dummy.cc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (davidlange6) [inline L1Trigger/L1TGlobal/src/Dummy.cc]: we don't want this file in your PR I think.

Comment (mulhearn) [inline L1Trigger/L1TGlobal/src/Dummy.cc]: You are right.

File: DataFormats/L1Trigger/interface/EtSumHelper.h (hunk 1 above)
Comment (mulhearn) [inline DataFormats/L1Trigger/interface/EtSumHelper.h]: You are right... meant to make this const correct after I checked the objects were const correct.  Dangers of late night coding!

**Comment by mulhearn** [comment]
Let me see if I can fix those issues.

**Comment by Martin-Grunewald** [comment]
@mulhearn @rekovic 

This PR is revealing, and is in itself, a problem:

On the one hand @rekovic  ensured me that for the EtSums we do NOT need a template
specialisation in the TriggerSummaryProducerAOD where we extract and pack up the
4-vector candidate content.
But on the other hand in this PR I see that the PHI coordinate is NOT taken from
the 4-vector candidate part, but rather from the hardware variable.

Hence: Can you please ensure that the 4-vector phi (and eta=0, mass=0, charge=0) is properly
set AND the above PR code is changed to pick phi from the 4-vector phi, so as to avoid HLT
coders to use hwPhi. (Remember l1extra was introduced to abstract away the hw coding and
use physics values in the CMS unit system.)

**Comment by mulhearn** [comment]
Indeed that is a mistake too... but I believe in upstream fix may be needed for the physical phi value:  investigating.

**Comment by rekovic** [comment]
@Martin-Grunewald  I did not appreciate the full shortcoming of using hardware vars at this moment.  Once the upstream fix for setting EtSum phi is applied, the template methods of TriggerSummaryProducerAOD will suffice and there is no need for specialisation.  We can concentrate on removing needs to use hw vars , and try to get that fix in.

**Comment by rekovic** [comment]
@mulhearn A fix for Muon charge() is also needed. Currently using hwCharge().

**Comment by mulhearn** [comment]
For technical reasons, closing this PR and opening #13319 which addresses all concerns above.

**Pull request closed without merging**